### DISCUSSION TITLE
Go 语言的代码生成机制 | Go 语言设计与实现 · /golang-code-gen

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#608] [Announcements]
https://draveness.me/golang-code-gen

 Go 语言的代码生成机制

**Comment by catundercar** [comment]
大哥，有时间可以出一期pprof的教程嘛。。

**Reply by ChunGang5** [reply]
https://zhuanlan.zhihu.com/p/396363069 看一下这篇帖子，粗略的将来一些参数

**Comment by draveness** [comment]
> 大哥，有时间可以出一期pprof的教程嘛。。

这个题目不错，不过教程可能不会写，这里应该会看看它是怎么做 Profile 的

**Comment by catundercar** [comment]
期待，小弟主要是看不太懂图，然后那些指数代表什么意义，不知道怎么入手，害。

**Comment by ghost** [comment]
大哥是陕西人吗

**Comment by shanyj** [comment]
大佬啥时候出下泛型的分析呀~

**Reply by draveness** [reply]
泛型实在是太丑了，最近都不想看 🤣

**Comment by blessli** [comment]
很多golang相关脚手架framework深度依赖元编程这一基础能力，来实现自动代码生成，比如protoc-gen-go-grpc
